### PR TITLE
OCPBUGS#17167:  Clarify the `Upgradeable` Operator condition

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Version(s): 4.10+

Issue: [OCPBUGS-17167](https://issues.redhat.com/browse/OCPBUGS-17167)

Link to docs preview: [OLM > Upgradeable condition](https://63029--docspreview.netlify.app/openshift-enterprise/latest/operators/understanding/olm/olm-operatorconditions.html#olm-upgradeable-operatorcondition_olm-operatorconditions)

QE review:
- [x] QE has approved this change.

Additional information: Follow up ticket to address QE feedback [OSDOCS-7513](https://issues.redhat.com//browse/OSDOCS-7513)

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
Looks good to me, but I want to describe more about `OperatorContition`. 
There is no `spec.conditions` field generated automatically as default. As follows,  only `spec.deployments` and   `spec.serviceAccounts` fields.
```console
$ oc get operatorcondition learn-operator.v0.0.2  -o yaml
apiVersion: operators.coreos.com/v2
kind: OperatorCondition
metadata:
  creationTimestamp: "2023-08-07T08:43:03Z"
  generation: 1
  labels:
    operators.coreos.com/learn.e2e-test-default-rs46yvij-dkm74: ""
  name: learn-operator.v0.0.2
  namespace: e2e-test-default-rs46yvij-dkm74
  ownerReferences:
  - apiVersion: operators.coreos.com/v1alpha1
    blockOwnerDeletion: false
    controller: true
    kind: ClusterServiceVersion
    name: learn-operator.v0.0.2
    uid: 123298fc-059e-48c8-98ca-2116340831f8
  resourceVersion: "217954"
  uid: b839016e-b4b0-4182-8c83-d668d845da72
spec:
  deployments:
  - learn-operator
  serviceAccounts:
  - learn-operator
  ```
The user can enable it by using `patch`, such as `$ oc patch operatorcondition learn-operator.v0.0.1 -p '{"spec": {"conditions": [{"type": "Upgradeable", "status": "False", "reason": "upgradeIsNotSafe", "message": "Disable the upgrade", "observedCondition":1, "lastUpdateTime":"2021-06-16T16:56:44Z","lastTransitionTime":"2021-06-16T16:56:44Z"}]}}', --type=merge`
```console
$ oc get operatorcondition learn-operator.v0.0.1 -o yaml
apiVersion: operators.coreos.com/v2
kind: OperatorCondition
metadata:
  creationTimestamp: "2023-08-07T08:42:21Z"
  generation: 2
  labels:
    operators.coreos.com/learn.e2e-test-default-rs46yvij-dkm74: ""
  name: learn-operator.v0.0.1
  namespace: e2e-test-default-rs46yvij-dkm74
  ownerReferences:
  - apiVersion: operators.coreos.com/v1alpha1
    blockOwnerDeletion: false
    controller: true
    kind: ClusterServiceVersion
    name: learn-operator.v0.0.1
    uid: 8afd2882-8d2a-469c-834b-d11930a0d055
  resourceVersion: "217760"
  uid: 34241680-38ac-40e5-b145-1475e9a1fd9f
spec:
  conditions:
  - lastTransitionTime: "2021-06-16T16:56:44Z"
    message: Disable the upgrade
    reason: upgradeIsNotSafe
    status: "False"
    type: Upgradeable
  deployments:
  - learn-operator
  serviceAccounts:
  - learn-operator
status:
  conditions:
  - lastTransitionTime: "2021-06-16T16:56:44Z"
    message: Disable the upgrade
    observedGeneration: 2
    reason: upgradeIsNotSafe
    status: "False"
    type: Upgradeable
```

**Review by awgreene — APPROVED** [review]
This looks good to me!

File: operators/understanding/olm/olm-operatorconditions.adoc [hunk 1]
```diff
@@ -17,3 +17,5 @@ include::modules/olm-supported-operatorconditions.adoc[leveloffset=+1]
 
 * xref:../../../operators/admin/olm-managing-operatorconditions.adoc#olm-operatorconditions[Managing Operator conditions]
 * xref:../../../operators/operator_sdk/osdk-generating-csvs.adoc#osdk-operatorconditions_osdk-generating-csvs[Enabling Operator conditions]
+* xref:../../../nodes/nodes/nodes-nodes-graceful-shutdown.adoc#nodes-nodes-graceful-shutdown[Managing graceful node shutdown]
+* xref:../../../applications/deployments/route-based-deployment-strategies.adoc#deployments-graceful-termination_route-based-deployment-strategies[Graceful termination]
```
Comment (wking) [inline operators/understanding/olm/olm-operatorconditions.adoc:21]: Possibly also worth a link to docs for PodDisruptionBudgets like [these][1]?

[1]: https://docs.openshift.com/container-platform/4.13/nodes/pods/nodes-pods-configuring.html#nodes-pods-pod-distruption-about_nodes-pods-configuring

Comment (michaelryanpeter) [reply operators/understanding/olm/olm-operatorconditions.adoc:21]: Updated. Thank you, @wking! I added the reference. I also had to remove the reference to "Managing graceful node shutdown", as the content is being removed from 4.13 and 4.14.

**Comment by michaelryanpeter** [comment]
@jianzhangbjz I am going to create a short procedure in a follow up PR on enabling Operator conditions. Here is the ticket for your reference: https://issues.redhat.com/browse/OSDOCS-7513. I have pulled it into the next sprint. Thank you for the feedback!

File: operators/understanding/olm/olm-operatorconditions.adoc [hunk 2]
```diff
@@ -17,3 +17,5 @@ include::modules/olm-supported-operatorconditions.adoc[leveloffset=+1]
 
 * xref:../../../operators/admin/olm-managing-operatorconditions.adoc#olm-operatorconditions[Managing Operator conditions]
 * xref:../../../operators/operator_sdk/osdk-generating-csvs.adoc#osdk-operatorconditions_osdk-generating-csvs[Enabling Operator conditions]
+* xref:../../../nodes/pods/nodes-pods-configuring.adoc#nodes-pods-configuring-pod-distruption-about_nodes-pods-configuring[Understanding how to use pod disruption budgets to specify the number of pods that must be up]
```
Comment (jeana-redhat) [inline operators/understanding/olm/olm-operatorconditions.adoc:20]: [Out of scope, food for thought] Is there a reason not to call this topic just "Using pod disruption budgets to specify the number of pods that must be up" to get it a leeeetle shorter?

**Review by jeana-redhat — APPROVED** [review]
LGTM

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**